Sources/StreamChat/APIClient/APIClient.swift

```diff
@@ -31,7 +31,7 @@ class APIClient {
     let decoder: RequestDecoder
     
     /// Used for reobtaining tokens when they expire and API client receives token expiration error
-    let tokenRefresher: (ClientError, @escaping () -> Void) -> Void
+    let tokenRefresher: (@escaping () -> Void) -> Void
```

Comment:
I would rather have a protocol for that so the API is self-explanatory. All token-related stuff from `APIClient` e.g. `isRefreshingToken/tokenRefreshConsecutiveFailures/maxTokenRefreshAttempts` should also be moved there IMHO.

Reply:
btw,`tokenRefreshConsecutiveFailures/maxTokenRefreshAttempts` is actually a retry strategy which is already encapsulated 🙂